Source/Core/Core/AchievementManager.cpp

```diff
@@ -1016,694 +520,461 @@ void AchievementManager::FilereaderClose(void* file_handle)
-                                                                       int value)
+void AchievementManager::HandleAchievementChallengeIndicatorShowEvent(
+    const rc_client_event_t* client_event)
 {
-  std::string username = Config::Get(Config::RA_USERNAME);
-  std::string api_token = Config::Get(Config::RA_API_TOKEN);
-  rc_api_submit_lboard_entry_request_t submit_request = {.username = username.c_str(),
-                                                         .api_token = api_token.c_str(),
-                                                         .leaderboard_id = leaderboard_id,
-                                                         .score = value,
-                                                         .game_hash = m_game_hash.data()};
-  rc_api_submit_lboard_entry_response_t submit_response = {};
-  ResponseType r_type =
-      Request<rc_api_submit_lboard_entry_request_t, rc_api_submit_lboard_entry_response_t>(
-          submit_request, &submit_response, rc_api_init_submit_lboard_entry_request,
-          rc_api_process_submit_lboard_entry_response);
-  rc_api_destroy_submit_lboard_entry_response(&submit_response);
-  if (r_type == ResponseType::SUCCESS)
+  if (Config::Get(Config::RA_BADGES_ENABLED) &&
+      AchievementManager::GetInstance().m_unlocked_badges.contains(client_event->achievement->id))
   {
-    INFO_LOG_FMT(ACHIEVEMENTS, "Submitted leaderboard ID {}.", leaderboard_id);
+    AchievementManager::GetInstance()
+        .m_active_challenges[client_event->achievement->badge_name] = DecodeBadgeToOSDIcon(
+        AchievementManager::GetInstance().m_unlocked_badges[client_event->achievement->id].badge);
   }
-  else
-  {
-    WARN_LOG_FMT(ACHIEVEMENTS, "Failed to submit leaderboard ID {}.", leaderboard_id);
-  }
-  return r_type;
-}
-
-AchievementManager::ResponseType
-AchievementManager::PingRichPresence(const RichPresence& rich_presence)
-{
-  std::string username = Config::Get(Config::RA_USERNAME);
-  std::string api_token = Config::Get(Config::RA_API_TOKEN);
-  rc_api_ping_request_t ping_request = {.username = username.c_str(),
-                                        .api_token = api_token.c_str(),
-                                        .game_id = m_game_id,
-                                        .rich_presence = rich_presence.data()};
-  rc_api_ping_response_t ping_response = {};
-  ResponseType r_type = Request<rc_api_ping_request_t, rc_api_ping_response_t>(
-      ping_request, &ping_response, rc_api_init_ping_request, rc_api_process_ping_response);
-  rc_api_destroy_ping_response(&ping_response);
-  return r_type;
 }
 
-void AchievementManager::DisplayWelcomeMessage()
+void AchievementManager::HandleAchievementChallengeIndicatorHideEvent(
+    const rc_client_event_t* client_event)
 {
-  std::lock_guard lg{m_lock};
-  PointSpread spread = TallyScore();
-  if (Config::Get(Config::RA_HARDCORE_ENABLED))
+  if (AchievementManager::GetInstance().m_active_challenges.contains(
```

Comment:
You should be able to do:

`m_active_challenges.erase(client_event->achievement->badge_name);`

No need for the extra lookup?

Reply:
I could have _sworn_ I looked this up and was told erase throws if the element doesn't exist, but that's not what the documentation is telling me right now.